spec/models/order_spec.rb

```diff
@@ -0,0 +1,51 @@
+require "rails_helper"
+
+RSpec.describe Order, type: :model do
+  describe "Associations" do
+    it {
+      should belong_to(:user)
+      should belong_to(:discount).optional(:true)
+      should belong_to(:address).optional(:true)
+      should have_many(:order_details).dependent(:destroy)
+      should have_many(:book_details).through(:order_details)
+    }
+  end
+
+  describe "Enums" do
+    it {
+      should define_enum_for(:status)
+        .with_values(pending: 0, accepted: 1, rejected: 2)
+        .with_suffix
+    }
+  end
+
+  describe "#send_mail_approve" do
+    order = FactoryBot.create(:order)
+    it "send mail approve" do
+      ActionMailer::Base.deliveries.should be_empty
+      mail = order.send_mail_approve
+      expect(mail.to).to eq [order.user.email]
+      expect(mail.subject).to eq "Order has been confirmed"
+      ActionMailer::Base.deliveries.should_not be_empty
+    end
+  end
+
+  describe "#send_mail_reject" do
```

Comment:
phân biệt describer, context và it

Reply:
- describe là từ khóa khai báo 1 example group, bên trong có chứa các example hoặc nested group
- context là từ khóa khai bao nested group là 1 tập hợp các example nằm bên trong 1 từ khóa describe
- it là từ khóa khai báo 1 example, tương đương với 1 test